src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.account.ActiveAccount;
+import seedu.address.model.account.entry.ExpenseDescriptionContainsKeywordsPredicate;
+import seedu.address.model.account.entry.RevenueDescriptionContainsKeywordsPredicate;
+
+
+public class FindCommand extends Command {
+    public static final String COMMAND_WORD = "find";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all entries whose descriptions contain any of "
+        + "the specified keywords (case-insensitive) and displays them as lists with index numbers.\n"
+        + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+        + "Example 1: " + COMMAND_WORD + "k/canvas cases\n"
+        + "Example 2: " + COMMAND_WORD + "k/canvas c/expense\n"
+        + "Example 3: " + COMMAND_WORD + "k/cases c/revenue\n";
+
+    private final ExpenseDescriptionContainsKeywordsPredicate expensePredicate;
+    private final RevenueDescriptionContainsKeywordsPredicate revenuePredicate;
+
+    /**
+     * Creates a FindCommand with a given predicate to find revenues with the given keywords.
+     */
+    public FindCommand(RevenueDescriptionContainsKeywordsPredicate predicate) {
+        this.expensePredicate = null;
+        this.revenuePredicate = predicate;
+    }
+
+    /**
+     * Creates a FindCommand with a given predicate to find expenses with the given keywords.
+     */
+    public FindCommand(ExpenseDescriptionContainsKeywordsPredicate predicate) {
+        this.expensePredicate = predicate;
+        this.revenuePredicate = null;
+    }
+
+    /**
+     * Creates a FindCommand with a given predicate to find expenses and revenues with the given keywords.
+     */
+    public FindCommand(ExpenseDescriptionContainsKeywordsPredicate expensePredicate,
+                       RevenueDescriptionContainsKeywordsPredicate revenuePredicate) {
+        this.expensePredicate = expensePredicate;
+        this.revenuePredicate = revenuePredicate;
+    }
+
+    @Override
+    public CommandResult execute(Model model, ActiveAccount activeAccount) throws CommandException {
+        requireNonNull(model);
```

Comment:
This should include the activeAccount but I can change it once everyone finishes the command